src/systems/inter_mesh_projection.C

```diff
@@ -75,23 +75,18 @@ void InterMeshProjection::project_system_vectors()
   (*solution_vector_serial) = solution_vector;
 
   // Construct a MeshFunction for the solution
-  MeshFunction * mesh_func_solution =
-    new MeshFunction(from_system.get_equation_systems(), *solution_vector_serial,
-                     from_system.get_dof_map(), variables_vector);
+  std::unique_ptr<MeshFunction> mesh_func_solution(new MeshFunction(from_system.get_equation_systems(), *solution_vector_serial,
```

Comment:
The RAII here is much better than having to manually delete later, but you can use std::make_unique() to get rid of the call to new too.

Reply:
Actually ... do we need heap allocation here at all?  I had to use it in the original projection.C because we needed a separate MeshFunction for each variable, but now that we support that variables_vector and only need one MeshFunction, can't we just have the MeshFunction (and GradientMeshFunction) on the stack?

Reply:
> you can use std::make_unique() to get rid of the call to new too.

`libmesh_make_unique<T>` still, unless we officially allow C++14 now that you merged #2842?

Reply:
libmesh_make_unique still, you're right.

Reply:
Thinking about this some more: In theory our CI tests are still going to catch any C++14 code that slips into libMesh, despite the compiler flag request changes in #2842, because at least the minimum-clang test ends up falling back on C++11 when it hits that too-old-libstdc++ problem while testing code under clang++ C++14 settings.  But in practice I should probably double-check with @loganharbour that the CI migrations they've been working on aren't going to change anything in that regard.

My personal preference is leaning more and more in favor of us to start requiring C++17 support and using newer features by the end of the 2021, a year and a half after the new stable RHEL ecosystem started supporting it by default (and years after major compilers have been supporting it for power users), but I definitely want any libMesh standards requirement upgrade to be a collective reasoned decision, not an accident, and I want to get at least one more C++11-compatible release out first.

Reply:
> Actually ... do we need heap allocation here at all? I had to use it in the original projection.C because we needed a separate MeshFunction for each variable, but now that we support that variables_vector and only need one MeshFunction, can't we just have the MeshFunction (and GradientMeshFunction) on the stack?

This is for each system vector.

Reply:
Sure, but do you operate on all the vectors at the same time, so that you have to have all the corresponding MeshFunctions defined at the same time?  If each MeshFunction is getting used only within a single block then you might as well declare it on the stack in that block and let it go out of scope afterwards.